### PR TITLE
implement lgpio-lib for RpiGpioActuator & GpioColorLED

### PR DESCRIPTION
**Pull request by DanielDecker**
This PR changes the GPIO library for the RpiGpioActuator & GpioColorLED to lgpio.

*changed RpiGpioActuator & GpioColorLED to use lgpio instead of RPi.GPIO
+added type annotation for RpiGpioActuator, GpioColorLED, core.sensor and core.actuator to allow type checking (e. g. with mypy)
*GpioColorLED modified so it can be used as a single-channel dimmer

Difference to the RPi.GPIO implementation: after exiting sensor_reporter the GPIOs return to the state before sensor_reporter claimed them.
In a later PR I'll remove the ColorHSV class from the i2c.pwm file and use the one in utils.

Tested with Debian 12 (bookworm, kernel 6.6) with mqtt, home and openHab connector. This PR is ready to merge.

### CODE REVIEW

**Comment by rkoshak** [comment]
Do you want to make the changes to dependencies.txt in this PR too?

File: core/actuator.py [hunk 1]
```diff
@@ -17,81 +17,95 @@
...
-from core.utils import set_log_level
+from core import utils
+from core import connection
 
 class Actuator(ABC):
-    """Class from which all actuator capabilities must inherit. Is assumes there
-    is a "Topic" param and automatically registers to subscribe to that topic
-    with all of the passed in connections. A default implementation is provided
-    for all but the on_message method which must be overridden.
+    """ Class from which all actuator capabilities must inherit. Is assumes there
```
Comment (rkoshak) [inline core/actuator.py:26]: > Class from which all actuator capabilities must inherit. Is assumes there

Class from which all actuator capabilities must inherit. **It** assumes there

I know this type was in the original but I'm just now seeing it.

**Review by rkoshak — COMMENTED** [review]
Thanks for fixing all the *many* typos in the comments. I'm a horrible editor.

**Comment by DanielDecker** [comment]
> Do you want to make the changes to dependencies.txt in this PR too?

Yes, I just pushed a commit with this and the typo fix you mentioned.
Everything regarding this PR is done, it's ready for merge.

**Comment by rkoshak** [comment]
> Since my development environment has a spell checker, it's much easier to spot and correct the typos. :-)

I originally developed this using just vim and it shows.

**Pull request merged**